Description: Set an initial size for the xymondlog output
 Cherry-picked from the 4.3.18 upstream release.
 .
 Original commit message: "xymond: Set an initial size for the
 xymondlog output, to avoid realloc'ing the buffer too much. Fix a
 valgrind complaint. JC Cleaver"
 .
 Despite the harmless sounding commit message, this seems to fix a
 severe memory leak. Without this patch, valgrind reported the loss of
 about 12 MB every five minutes on a bigger installation.
Origin: http://sourceforge.net/p/xymon/code/7489/
Applied-Upstream: http://sourceforge.net/p/xymon/code/7489/

Index: xymon/xymond/xymond.c
===================================================================
--- xymon.orig/xymond/xymond.c	2015-02-13 16:50:42.000000000 +0100
+++ xymon/xymond/xymond.c	2015-02-13 17:21:42.000000000 +0100
@@ -3679,12 +3679,13 @@
 			}
 
 			xfree(msg->buf);
-			logdata = generate_outbuf(NULL, logfields, h, log, acklevel);
+			logdata = newstrbuffer(20480);
+			logdata = generate_outbuf(&logdata, logfields, h, log, acklevel);
 			addtobuffer(logdata, msg_data(log->message, 0));
 
 			msg->doingwhat = RESPONDING;
 			msg->buflen = STRBUFLEN(logdata);
-			msg->bufp = grabstrbuffer(logdata);
+			msg->bufp = msg->buf = grabstrbuffer(logdata);
 		}
 
 		clear_filter(logfilter);
